runbot_travis2docker/models/runbot_branch.py

```diff
@@ -0,0 +1,168 @@
+        match = re.search(
+            r'(ssh\:\/\/\w+@(?P<host>[a-zA-Z0-9_.-]+))(:{0,1})'
+            r'(?P<port>(\d+))?', ssh)
+        if not match:
+            return False
+        data = match.groupdict()
+        cmd.append(data['port'] or '22')
+        cmd.append(data['host'])
+        with open(os.path.expanduser('~/.ssh/known_hosts'), 'a+') as hosts:
+            new_keys = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE)
+            for key in new_keys.stdout:
+                if [line for line in hosts if (line.strip('\n') ==
+                                               key.strip('\n'))]:
+                    continue
+                hosts.write(key + '\n')
+        return True
+
+    @tools.ormcache('url', 'token')
+    def get_weblate_projects(self, url, token):
+        """Find all projects and components that are on weblate url.
+        The cache is handled by @tools.ormcache annotation if the url and token
+        were already searched"""
+        projects = []
+        items = []
+        page = 1
+        session = requests.Session()
+        session.headers.update({
+            'Accept': 'application/json',
+            'User-Agent': 'runbot_travis2docker',
+            'Authorization': 'Token %s' % token
+        })
+        while True:
+            response = session.get('%s/projects/?page=%s' % (url, page))
+            response.raise_for_status()
+            data = response.json()
+            items.extend(data['results'])
+            if not data['next']:
+                break
+            page += 1
+        for project in items:
+            components = []
+            page = 1
+            while True:
+                response = session.get('%s/projects/%s/components/?page=%s'
+                                       % (url, project['slug'], page))
+                response.raise_for_status()
+                data = response.json()
+                components.extend(data['results'] or [])
+                if not data['next']:
+                    break
+                page += 1
+            project['components'] = components
+            response = session.get('%s/projects/%s/repository'
+                                   % (url, project['slug']))
+            response.raise_for_status()
+            data = response.json()
+            if data['needs_commit']:
+                response = session.post('%s/projects/%s/repository'
+                                        % (url, project['slug']),
+                                        {'operation': 'commit'})
+                response.raise_for_status()
+            projects.append(project)
+        return projects
+
+    @api.model
+    def cron_weblate(self):
+        for branch in self.search([('uses_weblate', '=', True)]):
+            if (not branch.repo_id.weblate_token or
+                    not branch.repo_id.weblate_url or
+                    not branch.repo_id.weblate_ssh):
```

Comment:
I'd add those to the search, but doesn't matter so much I guess

Reply:
You are right!
A search is faster than a python validation.
Thanks!